### ISSUE TITLE
Implement List<T>.Capacity and List<T>.TrimExcess

### ISSUE THREAD

**Issue by TriEdgeAI** [bridgedotnet/Bridge#2693]
### Capacity 

https://msdn.microsoft.com/en-us/library/y52x03h2(v=vs.110).aspx
> Gets or sets the total number of elements the internal data structure can hold without resizing.
```cs
public int Capacity { get; set; }
```

### TrimExcess

https://msdn.microsoft.com/en-us/library/ms132207(v=vs.110).aspx
> Sets the capacity to the actual number of elements in the List<T>, if that number is less than a threshold value.
```cs
public void TrimExcess()
```

### Sample
https://deck.net/51ea168a229b316b0720202a90ebb8d0
```cs
public class Program
{
    public static void Main()
    {
        List<int> parts = new List<int>();
        
        Console.WriteLine(parts.Count);
        Console.WriteLine(parts.Capacity);
        
        parts.Add(1);
        parts.Add(2);
        parts.Add(3);
        parts.Add(4);
        parts.Add(5);

        Console.WriteLine(parts.Count);
        Console.WriteLine(parts.Capacity);
        
        parts.TrimExcess();
        Console.WriteLine(parts.Count);
        Console.WriteLine(parts.Capacity);
        
        parts.Clear();
        Console.WriteLine(parts.Count);
        Console.WriteLine(parts.Capacity);
    }
}
```

**Comment by michaelcheers** [comment]
Related to #338.

**Comment by TriEdgeAI** [comment]
<3

**Comment by geoffreymcgill** [comment]
`Capacity` was included in the **16.0.0-beta3** release.
